python-package/lightgbm/callback.py

```diff
@@ -55,7 +59,7 @@ def _format_eval_result(value: _EvalResultTuple, show_stdv: bool) -> str:
         return f"{value[0]}'s {value[1]}: {value[2]:g}"
     elif len(value) == 5:
         if show_stdv:
-            return f"{value[0]}'s {value[1]}: {value[2]:g} + {value[4]:g}"
+            return f"{value[0]}'s {value[1]}: {value[2]:g} + {value[4]:g}"  # type: ignore[misc]
```

Comment:
These eval result tuples all have the same first four attributes:

```text
(eval_name, metric_name, eval_result, is_higher_better)
```

In `lgb.cv()`, that `eval_result` is the mean of the metric across all cross-validation folds. In that case, these tuples can optionally contain a fifth value with the standard deviation of that metric across all cross-validation folds.

https://github.com/microsoft/LightGBM/blob/2fe2bf0675ef127a29f9bc49511fc0099cf3e140/python-package/lightgbm/engine.py#L513

`mypy` can't tell automatically that being inside this `if len(value) == 5:` block means that the tuple has 5 elements, and raises the following error on this line:

> callback.py:62: error: Tuple index out of range  [misc]

🤷🏻